plugins/services/src/js/containers/services/ServicesContainer.js

```diff
@@ -474,34 +475,21 @@ class ServicesContainer extends React.Component {
       return <RequestErrorMsg />;
     }
 
-    if (item instanceof Pod) {
-      return (
-        <PodDetail
-          actions={this.getActions()}
-          pod={item}>
-          {this.getModals(item)}
-          {children}
-        </PodDetail>
-      );
-    }
-
-    if (item instanceof Service) {
-      return (
-        <ServiceDetail
-          actions={this.getActions()}
-          errors={this.state.actionErrors}
-          clearError={this.clearActionError}
-          params={params}
-          routes={routes}
-          service={item}>
-          {this.getModals(item)}
-          {children}
-        </ServiceDetail>
-      );
-    }
-
     // Show Tree
-    if (item instanceof ServiceTree) {
+    const currentroutePath = reconstructPathFromRoutes(routes);
+    if (currentroutePath.startsWith('/services/overview')) {
+
+      if (!(item instanceof ServiceTree)) {
+        // Not found
+        return (
```

Comment:
Would be nice to start extracting this conditions block from the render and putting in a method function to make more clear (helps with unit test too) and possibly even re-use in case it's necessary.

e.g `notFoundComponent` inside you would have the render, if possible without passing any argument but accessing in props (only if possible).

```
// Not found
        return (
          <Page>
            <Page.Header breadcrumbs={<ServiceBreadcrumbs />} />
            <ServiceItemNotFound
              message={`The group with the ID of "${itemId}" could not be found.`} />
          </Page>
        );
```